contracts/modules/legacy/LegacyPooledStaking.sol

```diff
@@ -1247,7 +1243,7 @@ contract LegacyPooledStaking is IPooledStaking, MasterAwareV2 {
 
       productInitParams[index] = ProductInitializationParams(
         products[i], // productId
-        uint8(min(stakes[i] * 1e18 / deposit / 1e16, 100)), // weight (0-100)
+        uint8(stakes[i] * 1e18 / deposit / 1e16), // weight (0-100)
```

Comment:
If this is relying on the assumption that stakes[i] <= deposit let's state that assumption in a comment